spark-commons/src/test/scala/za/co/absa/spark/commons/utils/SchemaUtilsTest.scala

```diff
@@ -36,4 +36,15 @@ class SchemaUtilsTest extends AnyFunSuite with Matchers {
     assertResult(getParentPath("a.bcd"))("a")
     assertResult(getParentPath(""))("")
   }
+
+  test("Test splitPath") {
+    assertResult(List("a", "b", "c", "d", "e"))(splitPath("a.b.c.d.e"))
+    assertResult(List("a"))(splitPath("a"))
+    assertResult(List("a", "bcd"))(splitPath("a.bcd"))
+    assertResult(List("a", "bcd"))(splitPath("a.bcd."))
+    assertResult(List("", "a", "bcd"))(splitPath(".a.bcd"))
```

Comment:
```suggestion
    assertResult(List("", "a", "bcd"))(splitPath(".a.bcd"))
    assertResult(List("", "", "", "a", "bcd"))(splitPath("...a.bcd"))
    assertResult(List("a", "bcd"))(splitPath("a.bcd."))
    assertResult(List("", "", "a", "bcd"))(splitPath("a.bcd...")) // is this expected behavior?
```
worth to demonstrate the trailing `.` corner cases, too?

Reply:
I was wondering about the last case too (multiple dots and the end). Not sure which is the most logical/expected behavior. 🤔 Any opinions?

Reply:
Are we providing only the service of "path splitting"? With available information we cannot decide what is valid or invalid input: "...ab.cd" or "ab.cd...." or "ab...cd".

Reply:
I was thinking about possible approaches. Let's vote (with emoticons):
* Keep as it is 👍 
* Throw exception if there's an empty field name 🚀 
* Remove all empty field names from result list 🎉 

I still think that one closing dot (`.`) could still be accepted.

Reply:
My vote would be
```scala
def splitPath(path: String, keepEmptyFields: Boolean = true)
```
Had I had to choose from the above, I'd go 🎉 .

Reply:
Tests Passed.